### PR TITLE
minimal change to update to BOBA-fix zingolib

### PR DESCRIPTION
**Pull request by zancas**
I haven't run `cargo update --aggressive -p zingolib` yet, but I think these manifest changes are a necessary pre-requisite.

The patches are either obsolete in zingolib:

* orchard
* bellman

Or there's a conflict between zingolib and zingo-mobile:

* zcash_note_encryption
* zcash_primitives

Patches only respect the top level workspace (not patches specified in transitive dependencies)

I am unclear about `sodiumoxide` (part of why this is a draft nature of this PR).

Note:   Cargo.lock is not yet modified (in commit 1).

### CODE REVIEW

**Comment by juanky201271** [comment]
I can answer about `sodiumoxide`... I have to fork this only to be able to build for ios-emulator. You don't remember, but I did a PR for this repo & the maintainer closed all PR's and archived the repo. Because of that I had to use my fork...

File: rust/lib/src/lib.rs [hunk 1]
```diff
@@ -37,14 +37,19 @@ fn construct_uri_load_config(
...
 ) -> Result<(zingoconfig::ZingoConfig, u64), String> {
-    let server = construct_server_uri(Some(uri));
+    let lightwalletd_uri = construct_lightwalletd_uri(Some(uri));
 
-    let (mut config, latest_block_height) = match zingolib::load_clientconfig(server, None) {
-        Ok((c, h)) => (c, h),
+    let mut config = match zingolib::load_clientconfig(
+        lightwalletd_uri.clone(),
+        None,
+        zingoconfig::ChainType::Mainnet,
```
Comment (AloeareV) [inline rust/lib/src/lib.rs:45]: I'd like to get #276 in ASAP, to get a better solution than hard-coded mainnet...transaction fees are about to increase when zip317 lands, which we can bandaid by hardcoding a fee of the new min-fee (10000), and then maybe when we minor-release that we can get #276 in if it's ready? I'm mostly thinking out loud.

**Review by AloeareV — DISMISSED** [review]

**Review by fluidvanadium — DISMISSED** [review]
nice update

**Review by juanky201271 — APPROVED** [review]
If the tests agree... this should be merged. Great job.

**Pull request merged**